### PR TITLE
Add scons buildsystem to package.tools

### PR DESCRIPTION
**Pull request by PucklaMotzer09**
This adds the scons buildsystem so that it can be more easily used inside packages.

### CODE REVIEW

File: xmake/modules/package/tools/scons.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import("core.base.option")
+import("lib.detect.find_tool")
+
+-- build package
+function build(package, configs, opt)
+    opt = opt or {}
+    local buildir = opt.buildir or os.curdir()
+    local njob = opt.jobs or option.get("jobs") or tostring(math.ceil(os.cpuinfo().ncpu * 3 / 2))
+    local scons = assert(find_tool("scons"), "scons not found!")
+    local argv = {"-C", buildir, "-j", njob}
```
Comment (waruqi) [inline xmake/modules/package/tools/scons.lua:31]: Can you add verbose output?

```lua
if option.get("verbose") then
    table.insert("argv", ...) --  scons VERBOSE=1?
end
```

Comment (PucklaMotzer09) [reply xmake/modules/package/tools/scons.lua:31]: I looked at the output of ```scons -H``` and there seems to be no such thing as verbose output. ```-v``` is used for version displaying. At default scons outputs every invoked compiler call.

Comment (waruqi) [reply xmake/modules/package/tools/scons.lua:31]: try `scons VERBOSE=1`?

https://scons.org/doc/1.0.1/HTML/scons-user/x1884.html

Comment (PucklaMotzer09) [reply xmake/modules/package/tools/scons.lua:31]: This does not seem to change anything. I also see every compiler call. But we could leave it in anyway.

Comment (waruqi) [reply xmake/modules/package/tools/scons.lua:31]: ok, I merged it first. we can improve it later.

**Pull request merged**